lib/random_assign_utils.rb

```diff
@@ -0,0 +1,37 @@
+# frozen_string_literal: true
+
+class RandomAssignUtils
+  def self.user_tzinfo(user_id)
```

Comment:
I wonder if this method would be useful to have in core? It's a fair amount of logic and seems reusable.

Reply:
Sure, at the moment it's in dev-additions and I ported it over here.

Where would you see it defined?

```
User.tzinfo(user_id)
UserOption.tzinfo(user_id)
```

?

Reply:
I'd say `UserOption` to keep `User` shorter.